### PR TITLE
[custom] New check on bundle feature.xml file

### PR DESCRIPTION
**Pull request by Hilbrand**

### CODE REVIEW

**Comment by Hilbrand** [comment]
:tada: builds again. @wborn can I persuade you to have a look :smile:

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/KarafAddonFeatureCheck.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+            if (exclude.equals(featureName)) {
+                logger.debug("Ignore check on feature name {}", featureName);
+                return;
+            }
+        }
+        checkName(featureFile, featureName, fileText, featureXML, FEATURE_NAME_EXPRESSION,
+                name -> !featureName.equals(name), MSG_FEATURE_NAME_INVALID, FEATURE_SEARCH);
+    }
+
+    private String adapatedFeatureName(String artifactId) {
```
Comment (wborn) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/KarafAddonFeatureCheck.java:192]:
```suggestion
    private String adaptedFeatureName(String artifactId) {
```

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/KarafAddonFeatureCheck.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    }
+
+    private void checkFeatures(File featureFile, String artifactId, FileText fileText, Document featureXML) {
+        checkName(featureFile, artifactId, fileText, featureXML, FEATURES_NAME_EXPRESSION,
+                name -> !artifactId.equals(name.substring(0, name.indexOf('-'))), MSG_FEATURES_NAME_INVALID,
+                FEATURES_SEARCH);
+    }
+
+    private void checkFeature(File featureFile, String artifactId, FileText fileText, Document featureXML) {
+        final String featureName = adapatedFeatureName(artifactId);
```
Comment (wborn) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/KarafAddonFeatureCheck.java:180]:
```suggestion
        final String featureName = adaptedFeatureName(artifactId);
```

File: custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/KarafAddonFeatureCheckTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheckTest;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+
+/**
+ * Tests for {@link KarafAddonFeatureCheck}
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@RunWith(MockitoJUnitRunner.class)
```
Comment (wborn) [inline custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/KarafAddonFeatureCheckTest.java:35]: This is for JUnit 4 and I don't see any `@Mock`s?

```suggestion
```

**Review by wborn — CHANGES_REQUESTED** [review]
Thanks for adding this useful check @Hilbrand! It will certainly help with reducing issues in the features. I have few minor comments below.

**Review by wborn — APPROVED** [review]

**Pull request merged**